File: swift/swift/Sources/tutorial/Hud.swift
import SwiftGodot

class Hud: CanvasLayer {
    @Node("Message") var message: Label
    @Node("ScoreLabel") var scoreLabel: Label
    @Node("StartButton") var startButton: Button
    @Node("MessageTimer") var messageTimer: Timer

    @Signal var startGame: SimpleSignal

    func showMessage(text: String, fade: Bool) {
        self.message.text = text
        self.message.show()
        if fade {
            self.messageTimer.start()
        }
    }

    func updateScore(score: Float) {
        self.scoreLabel.text = String(score)
    }

    func showGameOver() async {
        self.showMessage(text: "Game Over", fade: true)
        await self.messageTimer.timeout.emitted
        self.showMessage(text: "Dodge The Creeps!", fade: false)
        await self.getTree()?.createTimer(timeSec: 1.0)?.timeout.emitted
        self.startButton.show()
    }

    override func _ready() {
        self.startButton.pressed.connect {
            self.onStartButtonPressed()
        }
        self.messageTimer.timeout.connect {
            self.onMessageTimerTimeout()
        }
    }

    private func onStartButtonPressed() {
        self.startButton.hide()
        self.startGame.emit()
    }

    private func onMessageTimerTimeout() {
        self.message.hide()
    }
}

File: csharp/Game/Hud.cs
using System.Globalization;
using System.Threading.Tasks;
using Godot;
using NodeGetterGenerators;
using Timer = Godot.Timer;

namespace tutorial;

[VerifyNodeGetters("HUD")]
public partial class Hud : CanvasLayer
{
    [Node("Message", cache: true)]
    private partial Label MessageLabel { get; }

    [Node("ScoreLabel")]
    private partial Label ScoreLabel { get; }

    [Node("StartButton")]
    private partial Button StartButton { get; }

    [Node("MessageTimer")]
    private partial Timer MessageTimer { get; }

    [Signal]
    public delegate void StartGameEventHandler();

    public override void _Ready()
    {
        base._Ready();
        StartButton.Pressed += OnStartButtonPressed;
        MessageTimer.Timeout += OnMessageTimerTimeout;
    }

    public void OnStartButtonPressed()
    {
        StartButton.Hide();
        EmitSignal(SignalName.StartGame);
    }

    public void OnMessageTimerTimeout()
    {
        MessageLabel.Hide();
    }

    public void ShowMessage(string text, bool fade)
    {
        MessageLabel.Text = text;
        MessageLabel.Show();
        if (fade)
            MessageTimer.Start();
    }

    public void UpdateScore(float score)
    {
        ScoreLabel.Text = score.ToString(CultureInfo.CurrentCulture);
    }

    public async Task ShowGameOver()
    {
        ShowMessage("Game Over", true);
        await ToSignal(MessageTimer, Timer.SignalName.Timeout);
        ShowMessage("Dodge The Creeps!", false);
        await ToSignal(GetTree().CreateTimer(1.0), Timer.SignalName.Timeout);
        StartButton.Show();
    }
}
